### PR TITLE
fix(governance): include proposal deposits in SPO stake

### LINKED ISSUE #1147: [Governance] SPO voting stake excludes active proposal deposits
## Problem

SPO voting power is calculated from `epoch_stake` without adding active governance proposal deposits delegated to the pool through their return accounts. This undercounts both individual pool voting power and total SPO voting stake. The discrepancy is reproducible against the Cardano node, so it is not just a Koios comparison or epoch-boundary timing issue.

Related release-testing task: #1145.

## Verified Environment

- Network: Preprod, network magic `1`.
- Yaci Store: `3.0.0-beta4-build1`, PostgreSQL, checked on 2026-09-16.
- Source reference: `e55dffb0e9e2af8c43aff8a502ebf56246fdfe5b` (`v3.0.0-beta4-build1`).
- References: Cardano node `11.0.1`, DB Sync `13.7.0.5`, and Koios Preprod.
- Ledger reference: `15eee8cb0adfe94f8e26a7057b734b85ff8ee94e`.
- This runtime reproduction was performed on the beta instance, not on the `2.0.3-rc1` binary. Applicability to the release branch should be checked as part of #1145.

### Direct node comparison at epoch 313

Yaci proposal status snapshot `E` uses the SPO stake stored under `epoch_stake.active_epoch = E + 1`. Comparing epoch 313 against the node's `spo-stake-distribution` query produces the following values, all in lovelace:

| Pool | Yaci stake | Node voting stake | Missing in Yaci |
|---|---:|---:|---:|
| `1e3105f23f2ac91b3fb4c35fa4fe301421028e356e114944e902005b` | 12074617487501 | 12076617487501 | 2000000000 |
| `2f29763b1ebee8e1c1eb4d2d3bfa3e884e7ccca85a964eb684927922` | 138492465211 | 140492465211 | 2000000000 |
| **Network total** | **1570522746348179** | **1570526746348179** | **4000000000** |

DB Sync confirms that each of these pools receives two eligible proposal deposits of `1000000000` lovelace through delegated return accounts. These are the only two pools with different amounts in this comparison. The total missing voting power is **4,000 ADA**.

### Historical example at epoch 254

Proposal: `4194bb86bf12c6c6a3762f1095fd33e8b7b9f33a90e178291020f516ea520dc9#0` (InfoAction, expired at epoch 254).

```text
Yaci total SPO Yes + No + Abstain:  594286276492900
Koios total SPO Yes + No + Abstain: 594386276492900
Difference:                          100000000000 lovelace = 100,000 ADA
```

DB Sync records a `100000000000` lovelace deposit and return address `stake_test1uq7dr0j4k53dy32l7hs36t6zvlxffxeulfrxr6u08dppqpqrs269n`, delegated to pool `88b9b5564aa6756b56fee46c344506c5607ea42217246476e2145160` before the snapshot. Separate differences in this proposal's No/Abstain buckets do not explain away the total-stake deficit.

## Reproduction

On a Preprod Yaci database retaining the historical snapshots:

```sql
SELECT epoch, gov_action_tx_hash, gov_action_index,
       (voting_stats::jsonb ->> 'spo_total_yes_stake')::numeric
       + (voting_stats::jsonb ->> 'spo_total_no_stake')::numeric
       + (voting_stats::jsonb ->> 'spo_total_abstain_stake')::numeric AS total_spo_stake
FROM yaci_store.gov_action_proposal_status
WHERE epoch = 254
  AND gov_action_tx_hash = '4194bb86bf12c6c6a3762f1095fd33e8b7b9f33a90e178291020f516ea520dc9'
  AND gov_action_index = 0;

SELECT pool_id, sum(amount) AS stake
FROM yaci_store.epoch_stake
WHERE active_epoch = 314
  AND pool_id IN (
    '1e3105f23f2ac91b3fb4c35fa4fe301421028e356e114944e902005b',
    '2f29763b1ebee8e1c1eb4d2d3bfa3e884e7ccca85a964eb684927922'
  )
GROUP BY pool_id;
```

The epoch 313 node evidence was collected with:

```bash
cardano-cli conway query spo-stake-distribution \
  --all-spos --testnet-magic 1 --socket-path "$CARDANO_NODE_SOCKET_PATH"
```

This command returns the current node snapshot. To repeat the node comparison in a later epoch, align the Yaci status/stake epochs with the node epoch and reconstruct the eligible deposits for that snapshot; do not compare a later live node distribution directly with epoch 313.

## Expected Behavior And Relevant Code

The ledger adds eligible proposal deposits to both the corresponding pool's stake and the total pool distribution in [`computeDRepDistr` / `addToPoolDistr`](https://github.com/IntersectMBO/cardano-ledger/blob/15eee8cb0adfe94f8e26a7057b734b85ff8ee94e/eras/conway/impl/src/Cardano/Ledger/Conway/Governance/DRepPulser.hs#L177). The resulting distribution is supplied to ratification. [`querySPOStakeDistr`](https://github.com/IntersectMBO/cardano-ledger/blob/15eee8cb0adfe94f8e26a7057b734b85ff8ee94e/libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs#L282) exposes the completed pulser distribution.

In Yaci, [`SPOVotingDataCollector.buildEpochAggregates` and `calculateSPOStakeByVote`](https://github.com/bloxbean/yaci-store/blob/e55dffb0e9e2af8c43aff8a502ebf56246fdfe5b/aggregates/governance-aggr/src/main/java/com/bloxbean/cardano/yaci/store/governanceaggr/service/SPOVotingDataCollector.java#L43) use `epoch_stake` directly. Proposal deposits are accounted for separately in DRep distribution, but not in this SPO path.

## Impact And Acceptance Criteria

The checked proposal status snapshots still matched DB Sync and Koios; no changed historical outcome was observed in that comparison. However, incorrect voting power can change the acceptance ratio for a proposal close to its SPO threshold.

- [ ] Include eligible active proposal deposits in individual SPO voting power and total SPO stake using the same ledger snapshot and lifecycle rules.
- [ ] Apply the adjusted amounts consistently to explicit votes and default votes, without changing ordinary staking/reward snapshot semantics or double-counting deposits.
- [ ] Cover return accounts with no pool delegation, deregistered accounts, pools absent from the distribution, multiple deposits for one pool, and proposal removal/refund boundaries.
- [ ] Add a near-threshold ratification regression test and verify epoch replay/rollback does not retain or duplicate deposit stake.
- [ ] Recheck the release branch and record whether a backport/retest is required for 2.0.3.

### PR DESCRIPTION
**Pull request by huylevt**
Fixes #1147 
## Summary

  Fix SPO voting stake aggregation so active governance proposal deposits are included in the stake of the pool delegated by each proposal return account.

## Changes

  - Include lifecycle-filtered active proposal deposits when collecting SPO voting data.
  - Resolve each proposal return address through the epoch stake snapshot to find its delegated pool.
  - Add proposal deposits to:
    - total SPO active stake
    - explicit SPO vote stake
    - default abstain / no-confidence delegated SPO stake
  - Add focused unit coverage for proposal deposits delegated to SPO pools, including zero-stake return-account membership.
  
## Pre-Merge Checklist

  - [ ] Sync preprod and test vote tallies (In progress)

### CODE REVIEW

File: aggregates/adapot/src/main/java/com/bloxbean/cardano/yaci/store/adapot/storage/impl/repository/EpochStakeRepository.java [hunk 1]
```diff
@@ -23,6 +23,9 @@ public interface EpochStakeRepository extends JpaRepository<EpochStakeEntity, Ep
     @Query("select e from EpochStakeEntity e where e.address = :address and e.epoch = :activeEpoch - 2")
     Optional<EpochStakeEntity> findByAddressAndActiveEpoch(String address, Integer activeEpoch);
 
+    @Query("select e from EpochStakeEntity e where e.address in :addresses and e.epoch = :activeEpoch - 2")
```
Comment (satran004) [inline aggregates/adapot/src/main/java/com/bloxbean/cardano/yaci/store/adapot/storage/impl/repository/EpochStakeRepository.java:26]: @huylevt Can you please check if this query is ok on mainnet data as I don't think we have any index on address ?

Comment (satran004) [reply aggregates/adapot/src/main/java/com/bloxbean/cardano/yaci/store/adapot/storage/impl/repository/EpochStakeRepository.java:26]: Ignore. It is already part of primary key.

**Review by satran004 — APPROVED** [review]
LGTM . Thanks @huylevt
